maven-model-builder/src/test/java/org/apache/maven/model/building/DefaultModelBuilderTest.java

```diff
@@ -0,0 +1,151 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import junit.framework.TestCase;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Parent;
+import org.apache.maven.model.Repository;
+import org.apache.maven.model.resolution.InvalidRepositoryException;
+import org.apache.maven.model.resolution.ModelResolver;
+import org.apache.maven.model.resolution.UnresolvableModelException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * @author Guillaume Nodet
+ */
+public class DefaultModelBuilderTest
+{
+
+    private static final String BASE1_ID = "thegroup:base1:pom";
+
+    private static final String BASE1 = "<project>\n" +
+            "  <modelVersion>4.0.0</modelVersion>\n" +
+            "  <groupId>thegroup</groupId>\n" +
+            "  <artifactId>base1</artifactId>\n" +
+            "  <version>1</version>\n" +
+            "  <packaging>pom</packaging>\n" +
+            "  <dependencyManagement>\n" +
+            "    <dependencies>\n" +
+            "      <dependency>\n" +
+            "        <groupId>thegroup</groupId>\n" +
+            "        <artifactId>base2</artifactId>\n" +
+            "        <version>1</version>\n" +
+            "        <type>pom</type>\n" +
+            "        <scope>import</scope>\n" +
+            "      </dependency>\n" +
+            "    </dependencies>\n" +
+            "  </dependencyManagement>\n" +
+            "</project>\n";
+
+    private static final String BASE2_ID = "thegroup:base2:pom";
+
+    private static final String BASE2 = "<project>\n" +
+            "  <modelVersion>4.0.0</modelVersion>\n" +
+            "  <groupId>thegroup</groupId>\n" +
+            "  <artifactId>base2</artifactId>\n" +
+            "  <version>1</version>\n" +
+            "  <packaging>pom</packaging>\n" +
+            "  <dependencyManagement>\n" +
+            "    <dependencies>\n" +
+            "      <dependency>\n" +
+            "        <groupId>thegroup</groupId>\n" +
+            "        <artifactId>base1</artifactId>\n" +
+            "        <version>1</version>\n" +
+            "        <type>pom</type>\n" +
+            "        <scope>import</scope>\n" +
+            "      </dependency>\n" +
+            "    </dependencies>\n" +
+            "  </dependencyManagement>\n" +
+            "</project>\n";
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testCycleInImports()
+            throws Exception
+    {
+        ModelBuilder builder = new DefaultModelBuilderFactory().newInstance();
+        assertNotNull( builder );
+
+        DefaultModelBuildingRequest request = new DefaultModelBuildingRequest();
```

Comment:
I think L93-L107 deserve to be extracted to method.

Reply:
I've moved the method in a new class.